### PR TITLE
More notebook testing - almost all python and pandas notebooks

### PR DESCRIPTION
**Pull request by colinmorris**
As part of CI testing, execute all but two python and pandas notebooks (exclusions noted in `notebooks/test.sh`).

Would still like to get some more coverage, but this seemed like a good place to checkpoint.

### CODE REVIEW

File: notebooks/test.sh [hunk 1]
```diff
@@ -6,23 +6,46 @@ set -x
...
-#     python3 clean.py $track
-#     python3 prepare_push.py $track
-#     python3 render.py $track
+pip3 install -r requirements.txt
 
-#     # TODO: Once basic checks are working we should also do the following here:
-#     # - download necessary datasets for this track
-#     # - nbconvert execute *all* exercise notebooks for the track
-# done
+TRACKS="embeddings pandas python"
```
Comment (dansbecker) [inline notebooks/test.sh]: Any reason not to include `machine_learning`

File: Jenkinsfile [hunk 2]
```diff
@@ -9,6 +9,8 @@ pipeline {
         GIT_COMMIT_AUTHOR = sh(returnStdout: true, script:"git log --format='%an' -n 1 HEAD").trim()
         GIT_COMMIT_SUMMARY = "`<https://github.com/Kaggle/docker-python/commit/${GIT_COMMIT}|${GIT_COMMIT_SHORT}>` ${GIT_COMMIT_SUBJECT} - ${GIT_COMMIT_AUTHOR}"
         SLACK_CHANNEL = "#learnops"
+        KAGGLE_KEY = credentials('KAGGLE_API_KEY')
```
Comment (dansbecker) [inline Jenkinsfile]: Where/when is the environment variable set?

File: notebooks/test.sh [hunk 3]
```diff
@@ -6,23 +6,46 @@ set -x
...
+    python3 clean.py $track
+    python3 prepare_push.py $track
+    # TODO: Fails due to failure when calling get_git_branch()
+    #python3 render.py $track
+done
 
-# For now, just run one notebook (which doesn't depend on any datasets)
-jupyter nbconvert --output "$TMP_DIR" --execute "$DIR/python/raw/ex_2.ipynb"
+TESTABLE_NOTEBOOK_TRACKS="python pandas"
+for track in $TESTABLE_NOTEBOOK_TRACKS
```
Comment (dansbecker) [inline notebooks/test.sh]: For my knowledge, what are the most relevant conditions needed for a track to be testable?

Just that it has tagged anywhere we expect it to raise exceptions and it has setup_data.sh?

File: Jenkinsfile (hunk 2 above)
Comment (colinmorris) [reply Jenkinsfile]: In the jenkins web UI under the "credentials" section.

File: notebooks/test.sh (hunk 3 above)
Comment (colinmorris) [reply notebooks/test.sh]: Yup, exactly.

**Pull request merged**